disputer/disputer.js

```diff
@@ -114,7 +114,7 @@ class Disputer {
           message: "Cannot dispute liquidation: not enough collateral (or large enough approval) to initiate dispute✋",
           sponsor: disputeableLiquidation.sponsor,
           liquidation: disputeableLiquidation,
-          error: new Error(error)
+          error: typeof error === "string" ? new Error(error) : error
```

Comment:
Looking at this code made me realize that we send Logger.errors from both `disputer/disputer.js` and `disputer/index.js` (same with `liquidator/index.js` and `liquidator/liquidator.js`)

This style is inconsistent with how we log errors in `monitors`, which only sends errors from the root `index.js`, but none of the `ContractMonitor/BalanceMonitor/CRMonitor/etc.` send Logger errors. Instead, they either send `Logger.warn`'s or they just throw errors that get caught by the `index.js`.

What are your opinions on how to handle this? I suggest we replace these `Logger.error's` with `Logger.warn`'s.

Reply:
This is a good question.

The way I see it, there are three classes of errors:
1. Errors where the process cannot move forward. Most of these are thrown by libraries out of our control or occur during initialization. Examples would include a node query that fails (and we can't force a retry) or an invalid config.
2. Errors where the process can move forward with other work despite the error. Examples are when the liquidator or disputer wants to send a transaction, but it fails for some reason.
3. Errors that don't cause the process to crash and aren't necessarily an actionable issue, but may signal a problem or just be useful for us to know about.

IMO:
1. Should be handled by a high-level catch that emits the logger error. 
2. Should be emitted by the inner process without triggering a crash, so other work isn't interrupted.
3. Should be a warning emitted by the inner process.

Reply:
My intuition says that the liquidator and disputer have instances of 2, whereas the monitors really don't, which is why they don't throw any `logger.error`s outside of the catch.

Reply:
I agree with classifying these liquidator/disputer catches under #2. However, if we ever want to add additional logic to the error handling beyond just the ternary checking whether the `error` is a string or an object, then this will create additional work.

Since these are smart contract revert errors, perhaps we should just assume that they are of type Error and just set `error: error`?

Reply:
Follow up: I made this change. No need to check if an error returned by a web3.send/web3.call is a string, we can assume it is an `Error`.